app/services/multiple_listing_image_service.rb

```diff
@@ -0,0 +1,136 @@
+# Multiple Listing Image Resizing and Remote Storage
+class MultipleListingImageService
+  attr_reader :errors
+
+  # TMP_DIR = 'tmp/images'.freeze
+  # REMOTE_IMAGE_PATH = 'images/listings'.freeze
+  # IMAGE_WIDTH = 768
+
+  def initialize(listing_hash)
+    @errors = []
+    @listing = listing_hash
+    @listing_id = listing_hash['Id']
+    @listing_images = listing_hash['Listing_Images']
+    @resized_listing_images = FileStorageService.find('images/listings')
+  end
+
+  def process_images
+    unless @listing_images
+      add_error(@errors, "No listing images provided for listing #{@listing_id}")
+      return self
+    end
+    Rails.logger.info("Processing Listing Images for #{@listing_id}")
+
+    @listing_images.each do |listing_image|
+      li_raw_image_url = listing_image['Image_URL']
+
+      li_raw_image_url.blank? && next
+
+      cache_string = get_cache_string(@listing_id, li_raw_image_url)
+      image_name = get_image_name(@listing_id, cache_string)
+      tmp_image_path = get_tmp_image_path(image_name)
+      remote_image_path = get_remote_image_path(image_name)
+      image_url = get_image_url(remote_image_path)
+
+      if !resized_image(image_name, @resized_listing_images)
+        Rails.logger.info("Resizing and uploading #{image_name}")
+        if resize_and_upload_image(li_raw_image_url, tmp_image_path, remote_image_path,
+                                   @listing_id)
+          create_or_update_listing_image(@listing_id, image_url, li_raw_image_url)
+        end
+      elsif !listing_image_current(@listing_id,
+                                   image_url) || ENV['FORCE_MULTIPLE_LISTING_IMAGE_UPDATE'].to_s.casecmp('true').zero?
+        create_or_update_listing_image(@listing_id, image_url, li_raw_image_url)
+      end
+    end
+    self
+  end
+end
+
+  private
+
+def get_cache_string(listing_id, raw_image_url)
+  cache_string ||= Digest::MD5.hexdigest(listing_id + raw_image_url)
+  cache_string
+end
+
+def get_image_name(listing_id, cache_string)
+  "#{listing_id}-#{cache_string}.jpg"
+end
+
+def get_remote_image_path(image_name)
+  "images/listings/#{image_name}"
+end
+
+def get_image_url(remote_image_path)
+  base_image_url = ENV['RESOURCE_URL']
+  "#{base_image_url}/#{remote_image_path}"
+end
+
+def get_tmp_image_path(image_name)
+  "tmp/images/#{image_name}"
+end
+
+def resized_image(image_name, resized_listing_images)
+  resized_listing_images.count { |file| file.key.end_with?(image_name) }.positive?
+end
+
+def listing_image_current(listing_id, image_url)
```

Comment:
It's good practice to append `?` to the end of names  for methods that only return boolean values, with no side effects.

Reply:
addressed